client/src/components/EditablePolyline.tsx

```diff
@@ -3,39 +3,149 @@ import { Polyline as CorePolyline, LeafletEvent, LatLng } from 'leaflet';
+  if (modified_unit_array === undefined || modified_unit_array.length !== 1) {
+    console.error("Invalid modified_unit_array.length");
+    return;
+  }
+  let modified_unit_point = modified_unit_array[0];
+
+  const modified_unit_point_index = unit.points.findIndex(unit_point => unit_point.lat === modified_unit_point.lat && unit_point.lon === modified_unit_point.lon);
+
+  let old_point = JSON.parse(JSON.stringify(unit.points[modified_unit_point_index]));
+  unit.points[modified_unit_point_index].lat = modified_poly_point.lat;
+  unit.points[modified_unit_point_index].lon = modified_poly_point.lng; // TODO setstate?
+  // TODO alt
+  gameService.sendRouteModify(unit, old_point, unit.points[modified_unit_point_index]);
+
+  console.log("Point modified.");  
+};
+
+function handle_point_inserted(polyline: CorePolyline, unit: Unit) { 
+  const is_new_point = (point: LatLng) => unit.points.some(unit_point => is_same_point(point, unit_point)) === false;
+
+  let latLngs = polyline.getLatLngs() as LatLng[];
+  let new_point_array = latLngs.filter(poly_point => is_new_point(poly_point));
+  if (new_point_array === undefined || new_point_array.length !== 1) {
+    console.error("Invalid new_point_array.length " + new_point_array.length);
+    return;
+  }
+  let new_point = new_point_array[0];
+  
+  const new_point_index = latLngs.findIndex(poly_point => poly_point.lat === new_point.lat && poly_point.lng === new_point.lng);            
+
+  // TODO Temporary solution, copy the previous point
+  let at = JSON.parse(JSON.stringify(unit.points[new_point_index]));
+  let new_unit_point = JSON.parse(JSON.stringify(unit.points[new_point_index - 1]));    
+  new_unit_point.lat = new_point.lat;
+  new_unit_point.lon = new_point.lng;
+  // TODO alt
+
+  gameService.sendRouteInsertAt(unit, new_unit_point, at);
+
+  console.log("New point created.");       
+};
+
+function handle_point_removed(polyline: CorePolyline, unit: Unit, appState: AppStateReturnType) { 
+  let latLngs = polyline.getLatLngs() as LatLng[];
+  if (latLngs.length === 0) {
+    // Note: Temporary limitation, UI can't add new WP without polyline now
+    console.log("Last WP cannot be removed!");        
+    redrawUnit(unit, appState);
+    return;
+  }
+  
+  const polyline_contains = (point: Point) => latLngs.some(poly_point => is_same_point(poly_point, point));
+  let new_point_array = unit.points.filter((unit_point: Point) => polyline_contains(unit_point) === false);
+  if (new_point_array === undefined || new_point_array.length !== 1) {
+    console.error("Invalid new_point_array.length");
+    return;
+  }
+  let new_point = new_point_array[0];
+
+  gameService.sendRouteRemove(unit, new_point);  
+  
+  console.log("Point removed");        
+}
+
+// TODO Hack to get around polyline is not recreated when state is changed
+// TODO Remove not handled 
+let unitMap: { [id: string] : Unit; } = {};
```

Comment:
Could you explain the problem you were trying to get around here? I'd like to better understand to see if there's a way we can get this to work without the workaround.

Reply:
EditablePolyline called every time the state changes, but leaflet will call onPolylineAdded only once after the first time EditablePolyline is created(until it is removed). So the unit onPolineAdded will see is the one passed in the first time. E.g. add new point will work only once as after that it will detect multiple new points added at the same time and rejects the operation.

_I didn't spend much time on this problem as I wanted to remove the old client asap so no one else starts working on it like I did._

Reply:
I pulled your branch down last night and tested it out, and it seems to be working as expected. So if the goal of this pr is to remove the old client, I'm fine with this getting merged as long as we're tracking on trello that this needs to be revisited and improved.